includes/amp-helper-functions.php

```diff
@@ -495,37 +545,37 @@ function amp_remove_endpoint( $url ) {
  * If there are known validation errors for the current URL then do not output anything.
  *
  * @since 1.0
- * @global WP_Query $wp_query
  */
 function amp_add_amphtml_link() {
-	global $wp_query;
-
 	/**
 	 * Filters whether to show the amphtml link on the frontend.
 	 *
-	 * @todo This filter's name is incorrect. It's not about adding a canonical link but adding the amphtml link.
 	 * @since 0.2
 	 */
-	if ( false === apply_filters( 'amp_frontend_show_canonical', true ) ) {
-		return;
+	if ( has_filter( 'amp_frontend_show_canonical' ) ) {
+		_deprecated_hook( 'amp_frontend_show_canonical', '1.6', 'amp_frontend_show_amphtml' );
 	}
 
-	$current_url = amp_get_current_url();
-
-	$amp_url = null;
-	if ( current_theme_supports( AMP_Theme_Support::SLUG ) ) {
-		if ( AMP_Theme_Support::is_paired_available() ) {
-			$amp_url = add_query_arg( amp_get_slug(), '', $current_url );
-		}
-	} elseif ( $wp_query instanceof WP_Query && ( $wp_query->is_singular() || $wp_query->is_posts_page ) && post_supports_amp( get_post( get_queried_object_id() ) ) ) {
-		$amp_url = amp_get_permalink( get_queried_object_id() );
+	if (
+		false === apply_filters( 'amp_frontend_show_amphtml', true ) ||
```

Comment:
I wonder if we should even consider this in this PR. I don't think that the filter is even the right way to suppress the `amphtml` link. If anything, the better way would seem to be filtering the output of `is_amp_available()`. This would then act at a lower level, similar to `amp_skip_post`. It would actually get to implement #2817.

Reply:
OK that sounds like a reasonable plan. I would recommend deprecating the `amp_frontend_show_canonical` filter, and filtering the output of `is_amp_available()` as suggested.

Reply:
I've reverted the changes here via https://github.com/ampproject/amp-wp/pull/4796/commits/ccddfb26078bb2d24d4b0d6ce20ef05ecf13ef00.